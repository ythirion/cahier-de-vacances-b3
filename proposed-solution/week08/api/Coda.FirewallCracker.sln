
Microsoft Visual Studio Solution File, Format Version 12.00
Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "FirewallCracker", "FirewallCracker\FirewallCracker.csproj", "{B0135C40-E548-4E71-A986-A5C93ED2FF7A}"
EndProject
Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "FirewallCracker.Tests", "FirewallCracker.Tests\FirewallCracker.Tests.csproj", "{28053F82-724E-46CD-AE74-4422CBC2682B}"
EndProject
Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "FirewallCracker.Core", "FirewallCracker.Core\FirewallCracker.Core.csproj", "{34E25038-7312-46B4-B9C5-815EC8C72BB2}"
EndProject
Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "FirewallCracker.Adapters", "FirewallCracker.Adapters\FirewallCracker.Adapters.csproj", "{B8A52DBF-128B-4B6A-AC51-914596BC2FE9}"
EndProject
Global
	GlobalSection(SolutionConfigurationPlatforms) = preSolution
		Debug|Any CPU = Debug|Any CPU
		Release|Any CPU = Release|Any CPU
	EndGlobalSection
	GlobalSection(ProjectConfigurationPlatforms) = postSolution
		{B0135C40-E548-4E71-A986-A5C93ED2FF7A}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
		{B0135C40-E548-4E71-A986-A5C93ED2FF7A}.Debug|Any CPU.Build.0 = Debug|Any CPU
		{B0135C40-E548-4E71-A986-A5C93ED2FF7A}.Release|Any CPU.ActiveCfg = Release|Any CPU
		{B0135C40-E548-4E71-A986-A5C93ED2FF7A}.Release|Any CPU.Build.0 = Release|Any CPU
		{28053F82-724E-46CD-AE74-4422CBC2682B}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
		{28053F82-724E-46CD-AE74-4422CBC2682B}.Debug|Any CPU.Build.0 = Debug|Any CPU
		{28053F82-724E-46CD-AE74-4422CBC2682B}.Release|Any CPU.ActiveCfg = Release|Any CPU
		{28053F82-724E-46CD-AE74-4422CBC2682B}.Release|Any CPU.Build.0 = Release|Any CPU
		{34E25038-7312-46B4-B9C5-815EC8C72BB2}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
		{34E25038-7312-46B4-B9C5-815EC8C72BB2}.Debug|Any CPU.Build.0 = Debug|Any CPU
		{34E25038-7312-46B4-B9C5-815EC8C72BB2}.Release|Any CPU.ActiveCfg = Release|Any CPU
		{34E25038-7312-46B4-B9C5-815EC8C72BB2}.Release|Any CPU.Build.0 = Release|Any CPU
		{B8A52DBF-128B-4B6A-AC51-914596BC2FE9}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
		{B8A52DBF-128B-4B6A-AC51-914596BC2FE9}.Debug|Any CPU.Build.0 = Debug|Any CPU
		{B8A52DBF-128B-4B6A-AC51-914596BC2FE9}.Release|Any CPU.ActiveCfg = Release|Any CPU
		{B8A52DBF-128B-4B6A-AC51-914596BC2FE9}.Release|Any CPU.Build.0 = Release|Any CPU
	EndGlobalSection
EndGlobal
