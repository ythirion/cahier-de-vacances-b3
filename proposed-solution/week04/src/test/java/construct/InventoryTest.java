package construct;

import org.approvaltests.Approvals;
import org.approvaltests.combinations.CombinationApprovals;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class InventoryTest {
    @Test
    void classicUnitTest() {
        // Arrange
        var artefact = new Artefact("Mirror Shard", 0, 0);
        var constructInventory = new ConstructInventory(new Artefact[]{artefact});

        // Act
        constructInventory.updateSimulation();

        // Assert
        assertThat(artefact.getTimeToLive()).isEqualTo(-1);
        assertThat(artefact.getIntegrity()).isEqualTo(0);
    }

    @Test
    void singleApprovalTestForExample() {
        var artefact = new Artefact("Mirror Shard", 0, 0);
        var constructInventory = new ConstructInventory(new Artefact[]{artefact});

        constructInventory.updateSimulation();

        // No more assertions here, just use Approvals
        Approvals.verify(artefact);
    }

    @Test
    void approvalTests() {
        CombinationApprovals.verifyAllCombinations(
                this::updateSimulation,
                new String[]{
                        "Mirror Shard",
                        "Aged Signal",
                        "Backdoor Pass to TAFKAL80ETC Protocol",
                        "Sulfuras Core Fragment"
                },
                new Integer[]{-1, 0, 10, 11, 51},
                new Integer[]{-1, 0, 1, 5, 49, 50}
        );
    }

    private String updateSimulation(String name, int integrity, int timeToLive) {
        var artefact = new Artefact(name, integrity, timeToLive);
        var constructInventory = new ConstructInventory(new Artefact[]{artefact});
        constructInventory.updateSimulation();

        return artefact.toString();
    }
}