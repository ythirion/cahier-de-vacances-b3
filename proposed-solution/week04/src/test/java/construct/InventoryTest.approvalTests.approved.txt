[Mirror Shard, -1, 0] => Mirror Shard, -1, -1 
[Mirror Shard, -1, 1] => Mirror Shard, -1, 0 
[Mirror Shard, -1, 49] => Mirror Shard, -1, 48 
[Mirror Shard, -1, 50] => Mirror Shard, -1, 49 
[Mirror Shard, 0, 0] => Mirror Shard, 0, -1 
[Mirror Shard, 0, 1] => Mirror Shard, 0, 0 
[Mirror Shard, 0, 49] => Mirror Shard, 0, 48 
[Mirror Shard, 0, 50] => Mirror Shard, 0, 49 
[Mirror Shard, 11, 0] => Mirror Shard, 9, -1 
[Mirror Shard, 11, 1] => Mirror Shard, 10, 0 
[Mirror Shard, 11, 49] => Mirror Shard, 10, 48 
[Mirror Shard, 11, 50] => Mirror Shard, 10, 49 
[Aged Signal, -1, 0] => Aged Signal, 1, -1 
[Aged Signal, -1, 1] => Aged Signal, 0, 0 
[Aged Signal, -1, 49] => Aged Signal, 0, 48 
[Aged Signal, -1, 50] => Aged Signal, 0, 49 
[Aged Signal, 0, 0] => Aged Signal, 2, -1 
[Aged Signal, 0, 1] => Aged Signal, 1, 0 
[Aged Signal, 0, 49] => Aged Signal, 1, 48 
[Aged Signal, 0, 50] => Aged Signal, 1, 49 
[Aged Signal, 11, 0] => Aged Signal, 13, -1 
[Aged Signal, 11, 1] => Aged Signal, 12, 0 
[Aged Signal, 11, 49] => Aged Signal, 12, 48 
[Aged Signal, 11, 50] => Aged Signal, 12, 49 
[Backdoor Pass to TAFKAL80ETC Protocol, -1, 0] => Backdoor Pass to TAFKAL80ETC Protocol, 0, -1 
[Backdoor Pass to TAFKAL80ETC Protocol, -1, 1] => Backdoor Pass to TAFKAL80ETC Protocol, 2, 0 
[Backdoor Pass to TAFKAL80ETC Protocol, -1, 49] => Backdoor Pass to TAFKAL80ETC Protocol, 0, 48 
[Backdoor Pass to TAFKAL80ETC Protocol, -1, 50] => Backdoor Pass to TAFKAL80ETC Protocol, 0, 49 
[Backdoor Pass to TAFKAL80ETC Protocol, 0, 0] => Backdoor Pass to TAFKAL80ETC Protocol, 0, -1 
[Backdoor Pass to TAFKAL80ETC Protocol, 0, 1] => Backdoor Pass to TAFKAL80ETC Protocol, 3, 0 
[Backdoor Pass to TAFKAL80ETC Protocol, 0, 49] => Backdoor Pass to TAFKAL80ETC Protocol, 1, 48 
[Backdoor Pass to TAFKAL80ETC Protocol, 0, 50] => Backdoor Pass to TAFKAL80ETC Protocol, 1, 49 
[Backdoor Pass to TAFKAL80ETC Protocol, 11, 0] => Backdoor Pass to TAFKAL80ETC Protocol, 0, -1 
[Backdoor Pass to TAFKAL80ETC Protocol, 11, 1] => Backdoor Pass to TAFKAL80ETC Protocol, 14, 0 
[Backdoor Pass to TAFKAL80ETC Protocol, 11, 49] => Backdoor Pass to TAFKAL80ETC Protocol, 12, 48 
[Backdoor Pass to TAFKAL80ETC Protocol, 11, 50] => Backdoor Pass to TAFKAL80ETC Protocol, 12, 49 
[Sulfuras Core Fragment, -1, 0] => Sulfuras Core Fragment, -1, 0 
[Sulfuras Core Fragment, -1, 1] => Sulfuras Core Fragment, -1, 1 
[Sulfuras Core Fragment, -1, 49] => Sulfuras Core Fragment, -1, 49 
[Sulfuras Core Fragment, -1, 50] => Sulfuras Core Fragment, -1, 50 
[Sulfuras Core Fragment, 0, 0] => Sulfuras Core Fragment, 0, 0 
[Sulfuras Core Fragment, 0, 1] => Sulfuras Core Fragment, 0, 1 
[Sulfuras Core Fragment, 0, 49] => Sulfuras Core Fragment, 0, 49 
[Sulfuras Core Fragment, 0, 50] => Sulfuras Core Fragment, 0, 50 
[Sulfuras Core Fragment, 11, 0] => Sulfuras Core Fragment, 11, 0 
[Sulfuras Core Fragment, 11, 1] => Sulfuras Core Fragment, 11, 1 
[Sulfuras Core Fragment, 11, 49] => Sulfuras Core Fragment, 11, 49 
[Sulfuras Core Fragment, 11, 50] => Sulfuras Core Fragment, 11, 50 
